notebook/services/api/api.yaml

```diff
@@ -654,12 +639,18 @@ definitions:
       id:
         type: string
         format: uuid
-      notebook:
-        type: object
-        properties:
-          path:
-            type: string
-            description: path to notebook
+      path:
+        type: string
+        description: "The path to the session.  For console sessions, the path 
+         contains the cwd of the kernel and the name of the console session, 
+         which may be a uuid"
+      type:
+        type: string;
+        description: session type
+        enum:
+          - notebook
+          - file
+          - console
```

Comment:
There is no filesystem knowledge in the current REST API. The Contents API path is exposed, but that need not map to any filesystem at all. The SessionManager asks the ContentsManager "What should the CWD of the kernel be for this file/notebook". The *client* of the session API does not have any way to influence this, and it shouldn't for notebooks. For notebooks, the information should continue to be just two pieces of information: the notebook path according to the ContentsManager and the kernel name, according to the KernelManager.